zod/src/types.ts

```diff
@@ -8,7 +8,8 @@ import type { ParseParams } from 'zod/lib/src/parser';
 
 export type Resolver = <T extends z.ZodSchema<any, any>>(
   schema: T,
-  options?: ParseParams,
+  schemaOptions?: ParseParams,
+  resolverOptions?: { mode: 'async' | 'sync' },
```

Comment:
to use this feature, i guess users will have to this:

`resolver: async (data, options) => await zodResolver(schema, options, 'async')`

Reply:
No they have to this:
```ts
useForm({
  resolver: zodResolver(schema, options, {mode: 'sync'})
})
```

 Or if you prefer, we can do the following:

```ts
interface Options {
  schema?: ParseParams;
  resolver?: { mode: 'async' | 'sync' }
}

export type Resolver = <T extends z.ZodSchema<any, any>>(
  schema: T,
  options?: Options
) => { ... }
```

So user will have to this:
```ts
useForm({
  resolver: zodResolver(schema, {
    resolver: { mode: 'sync' },
    schema: schemaZodOptions
  })
})
```
I think I prefer this one, what do you think ?

FYI: `async` is the default mode

Reply:
This looks great: 

```
useForm({
  resolver: zodResolver(schema, options, {mode: 'sync'})
})
```

Reply:
`options` is optional so it could be:

```ts
useForm({
  resolver: zodResolver(schema, undefined, { mode: 'sync' })
})
```